lib/rubygems/request_set/gem_dependency_api.rb

```diff
@@ -526,6 +552,7 @@ def gem_requires name, options # :nodoc:
     else
       @requires[name] << name
     end
+    raise "Unhandled gem options #{options.inspect}" unless options.empty?
```

Comment:
Hi @mpapis, I tested this with the gemfile in #1286 and this new error didn't occur.  I see this doesn't have a test, do you want me to write it?